File: Coding Club Beginner Projects/Lesson3/Updated Guessing Game/guess.py
"""
Even if no one askes this, this can also serve as a guideline as to what to talk about.
Possible questions and answers:
Q: What is "import random"?
A: Python has a lot of prebuilt code that was made by the creators.
    By typing "import random", import the code that is used to get random numbers,
    which allows us to use the functions in that code.

Q: How does random.randint() work?
A: random.randint() takes two numbers and input 
    and returns a random number within that range.
    The first number must be greater than the second.
"""

import random

print("Guessing Game!\n")

print('You will have six guesses.')

won = False

for i in range(6):
    user_guess = int(input('Please enter in what you think the number is: '))

    random_num = random.randint(1, 100)

    if user_guess == random_num:
        print("You guessed correctly!")
        won = True
        break
    else:
        print("You guessed wrong...")

if won:
    print("You won! Congratulations!")
else:
    print("You lost...")

File: Coding Club Beginner Projects/Lesson3/Updated Guessing Game/guess.js
/* 
 * Even if no one askes this, this can also serve as a guideline as to what to talk about.
 * Possible questions and answers:
 * Q: How does Math.random() work?
 * A: Math.random() returns a random number between 0 and 1.
 * This isn't very useful for our purposes, since we want a 
 * random number between 1 and 100. To do this, we simply multiply
 * the number that Math.random() gives us by 100. 
 * 
 * Q: What is "===" supposed to do? How is it different from "=="?
 * A: The triple equals not only compares the values of the variables,
 * but also checks if they are the same type.
 */

console.log("Guessing Game!\n\n");

console.log("You will have six tries to guess the number.");

var won = false;

for (var i = 0; i < 6; i++)
{
    var randomNum = Math.random() * 100;

    var userGuess = prompt("Please enter in your guess: ") * 1;

    if (userGuess === randomNum)
    {
        console.log("You guessed correctly!\n");
        won = true;
        break;
    }
    else
    {
        console.log("You guessed wrong...");
    }
}

if (won)
{
    console.log("You won! Congratulations!");
}
else
{
    console.log("You lost...");
}